ceph/rbd/rbd.py

```diff
@@ -0,0 +1,121 @@
+import datetime
+import json
+from time import sleep
+
+from ceph.ceph import CommandFailed
+from tests.rbd.exceptions import CreateFileError, ImportFileError
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class Rbd:
+    def __init__(self, nodes):
+
+        # Identifying Client node
+        for node in nodes:
+            if node.role == "client":
+                self.ceph_client = node
+
+    def check_pool_exists(self, pool_name: str) -> bool:
+        """
+        recursively checks if the specified pool exists in the cluster
+        Args:
+            pool_name: Name of the pool to be checked
+
+        Returns:  True -> pass, False -> fail
+
+        """
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=200)
+        while end_time > datetime.datetime.now():
+            out = self.exec_cmd(cmd="ceph df -f json", output=True)
+            existing_pools = json.loads(out)
+            if pool_name not in [ele["name"] for ele in existing_pools["pools"]]:
+                log.error(
+                    f"Pool:{pool_name} not populated yet\n"
+                    f"sleeping for 2 seconds and checking status again"
+                )
+                sleep(2)
+            else:
+                log.info(f"pool {pool_name} exists in the cluster")
+                return True
+        log.info(f"pool {pool_name} does not exist on cluster")
+        return False
+
+    def exec_cmd(self, **kw):
```

Comment:
The return value are not consistent. Could we revisit it?

Reply:
done